### PR TITLE
store,snappy: add an Assertion method to retrieve a single assertion from the assertion service

### PR DESCRIPTION
**Pull request by pedronis**
This adds an Assertion method to retrieve a single assertion from the assertion service.

It also cleans up a few places that were calling the store repo "snap" which was confusing I think.

### CODE REVIEW

File: snappy/snapp_test.go [hunk 1]
```diff
@@ -240,15 +240,15 @@ func (s *SnapTestSuite) TestUbuntuStoreRepositoryUpdates(c *C) {
 	var err error
 	s.storeCfg.BulkURI, err = url.Parse(mockServer.URL + "/updates/")
 	c.Assert(err, IsNil)
-	snap := store.NewUbuntuStoreSnapRepository(s.storeCfg, "")
```
Comment (mvo5) [inline snappy/snapp_test.go]: Woah, thanks for doing this. Make me wonder how this ended up being called snap in the first place. Oh well!

File: store/snap_remote_repo.go [hunk 2]
```diff
@@ -176,10 +203,13 @@ func (s *SnapUbuntuStoreRepository) setUbuntuStoreHeadersWithToken(req *http.Req
 	if s.storeID != "" {
 		req.Header.Set("X-Ubuntu-Store", s.storeID)
 	}
+}
 
-	if token != nil {
-		req.Header.Set("Authorization", oauth.MakePlaintextSignature(token))
-	}
+// small helper that sets the correct http headers for a store request including auth
+func (s *SnapUbuntuStoreRepository) setStoreReqHeaders(req *http.Request, accept string) {
```
Comment (mvo5) [inline store/snap_remote_repo.go]: :heart:

**Comment by mvo5** [comment]
Looks fine. Thanks! I guess at some (later) point will also have to teach the fake store in the integration tests about assertions(?).

**Comment by pedronis** [comment]
@mvo5 yes, is one of the todos in my card

File: store/snap_remote_repo.go [hunk 3]
```diff
@@ -148,26 +174,27 @@ func NewUbuntuStoreSnapRepository(cfg *SnapUbuntuStoreConfig, storeID string) *S
...
 }
 
-// small helper that sets the correct http headers for the ubuntu store
-func (s *SnapUbuntuStoreRepository) setUbuntuStoreHeaders(req *http.Request) {
-	var token *oauth.Token
-	ssoToken, _ := ReadStoreToken()
-	if ssoToken != nil {
-		token = &ssoToken.Token
+// setAuthHeader sets the authorization header
+func (s *SnapUbuntuStoreRepository) setAuthHeader(req *http.Request, token *StoreToken) {
```
Comment (emgee) [inline store/snap_remote_repo.go]: Just a passing comment really ... but *to me* setAuthHeader suggests it's setting something on the receiver, i.e. the store repo, but I think it only configures the `req` arg, right? Something like configureReqAuthHeader() might read a little better.

File: store/snap_remote_repo.go [hunk 4]
```diff
@@ -375,3 +401,55 @@ var download = func(name string, w io.Writer, req *http.Request, pbar progress.M
...
+	req, err := http.NewRequest("GET", url.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	ssoToken, err := ReadStoreToken()
+	if err != nil {
+		s.setAuthHeader(req, ssoToken)
+	}
+	req.Header.Set("Accept", asserts.MediaType)
```
Comment (emgee) [inline store/snap_remote_repo.go]: Can this block be replaced with a call to setStoreReqHeaders?

File: store/snap_remote_repo.go [hunk 5]
```diff
@@ -375,3 +401,55 @@ var download = func(name string, w io.Writer, req *http.Request, pbar progress.M
...
+
+	client := &http.Client{}
```
Comment (emgee) [inline store/snap_remote_repo.go]: Do we want to create a new client for every request?

**Comment by emgee** [comment]
lgtm, just a couple of comments.

File: store/snap_remote_repo.go (hunk 4 above)
Comment (pedronis) [inline store/snap_remote_repo.go]: @emgee kind of, we would set a lot more headers that aren't relevant to the assertion service

**Comment by pedronis** [comment]
retest this please

**Pull request merged**